src/core/tests/frontend/paddle/test_models/gen_scripts/generate_gather_nd.py

```diff
@@ -0,0 +1,76 @@
+#
+# gather_nd paddle model generator
+#
+import numpy as np
+from save_model import saveModel
+import paddle
+import sys
+
+
+def gather_nd(name: str, x, y):
+    paddle.enable_static()
+
+    with paddle.static.program_guard(paddle.static.Program(), paddle.static.Program()):
+        data = paddle.static.data(name="x", shape=x.shape, dtype=x.dtype)
+        index = paddle.static.data(name="index", shape=y.shape, dtype=y.dtype)
+        out = paddle.gather_nd(data, index)
+
+        cpu = paddle.static.cpu_places(1)
+        exe = paddle.static.Executor(cpu[0])
+        # startup program will call initializer to initialize the parameters.
+        exe.run(paddle.static.default_startup_program())
+        outs = exe.run(feed={"x": x, "index": y}, fetch_list=[out])
+
+        saveModel(
+            name,
+            exe,
+            feedkeys=["x", "index"],
+            fetchlist=[out],
+            inputs=[x, y],
+            outputs=[outs[0]],
+            target_dir=sys.argv[1],
+        )
+
+    return outs[0]
+
+
+def main():
+    tests_cases = [
+        "float32",
+        "int32",
+        "int64",
+    ]
+    x_shape = (10, 20)
+
+    for test in tests_cases:
+        x = np.random.rand(*x_shape).astype(test)
+        y = np.array([[0, 1], [1, 1]]).astype("int32")
+        gather_nd("gather_nd_" + test, x, y)
+
+    x = np.random.rand(*x_shape).astype("float32")
+    y = np.array([[], []]).astype("int32")
+    gather_nd("gather_nd_empty", x, y)
+
+    y = np.array([[1], [2]]).astype("int32")
+    gather_nd("gather_nd_low_index", x, y)
+
+    x_shape = (5, 2, 3, 1, 10)
+    x = np.random.rand(*x_shape).astype("float32")
+    y = np.array(
+        [
+            [0, 0, 1, 0, 4],
+            [4, 0, 0, 0, 1],
+            [0, 1, 2, 0, 2],
+            [0, 1, 2, 0, 6],
+            [1, 1, 2, 0, 3],
+        ]
+    ).astype("int32")
+    gather_nd("gather_nd_high_rank1", x, y)
+
+    index = np.vstack([np.random.randint(0, s, size=100) for s in x_shape]).T
```

Comment:
please also remove 'np.vstack' here(we may use 'np.array' here just as 'gather_nd_high_rank1').

Reply:
In fact, the following way of writing does not work.

```python
y = np.array([np.random.randint(0, s, size=2) for s in x_shape]).astype("int32").T
gather_nd("gather_nd_high_rank1", x, y)
```

When I modify it to be written as follows, it is ok

```python
    y = np.array(
        [
            [np.random.randint(0, s) for s in x_shape],
            [np.random.randint(0, s) for s in x_shape],
        ]
    ).astype("int32")
    gather_nd("gather_nd_high_rank1", x, y)
```